Jumpscale/servers/email_forwarder/gsmtpd/server.py

```diff
@@ -0,0 +1,224 @@
+#!/usr/bin/env python
+# encoding: utf-8
+
+"""
+An RFC 2821 smtp proxy server port from Python Standrad Library for Gevent usage
+"""
+
+# Overview:
+#
+# This file implements the minimal SMTP protocol as defined in RFC 821.  It
+# has a hierarchy of classes which implement the backend functionality for the
+# smtpd.  A number of classes are provided:
+#
+#   SMTPServer - the base class for the backend.  Raises NotImplementedError
+#   if you try to use it.
+#
+#   DebuggingServer - simply prints each message it receives on stdout.
+#
+#   PureProxy - Proxies all messages to a real smtpd which does final
+#   delivery.  One known problem with this class is that it doesn't handle
+#   SMTP errors from the backend server at all.  This should be fixed
+#   (contributions are welcome!).
+#
+# Please note that this script requires Python 2.7
+#
+# Author: Barry Warsaw <barry@python.org>
+#         Meng Zhuo <mengzhuo1203@gmail.com>
+# TODO:
+#
+# - support mailbox delivery
+# - alias files
+# - ESMTP
+# - handle error codes from the backend smtpd
+
+import logging
+logger = logging.getLogger(__name__)
+
+from collections import UserDict
+
+from gevent import socket, monkey, Timeout
+monkey.patch_all(thread=False)
```

Comment:
I think Jumpscale should handle the monkey patching.

Reply:
I am not sure what do you mean by that. As far as i know JS does not does not do any monkey patching in its loading and bootstrap code.